src/ui/server/shared-validation/number-hyphen-and-spaces-only/index.test.ts

```diff
@@ -0,0 +1,114 @@
+  const MINIMUM = 6;
+  const MAXIMUM = 6;
+
+  const mockBody = {
+    [FIELD_ID]: '11-22-33',
+  } as RequestBody;
+
+  describe('when a value is not provided', () => {
+    it('should return validation errors', () => {
+      const mockEmptyBody = {};
+
+      const result = numberHyphenSpacesOnlyValidation(mockEmptyBody, FIELD_ID, mockErrorMessagesObject, mockErrors, MINIMUM, MAXIMUM);
+
+      const expected = generateValidationErrors(FIELD_ID, mockErrorMessagesObject.IS_EMPTY, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the provided value has a letter', () => {
+    it('should return validation errors"', () => {
+      mockBody[FIELD_ID] = '11-22-3E';
+
+      const result = numberHyphenSpacesOnlyValidation(mockBody, FIELD_ID, mockErrorMessagesObject, mockErrors, MINIMUM, MAXIMUM);
+
+      const expected = generateValidationErrors(FIELD_ID, mockErrorMessagesObject.INCORRECT_FORMAT, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the provided value has a special character', () => {
+    it('should return validation errors"', () => {
+      mockBody[FIELD_ID] = '11-22-3!';
+
+      const result = numberHyphenSpacesOnlyValidation(mockBody, FIELD_ID, mockErrorMessagesObject, mockErrors, MINIMUM, MAXIMUM);
+
+      const expected = generateValidationErrors(FIELD_ID, mockErrorMessagesObject.INCORRECT_FORMAT, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the provided value is below minimum length', () => {
+    it('should return validation errors', () => {
+      mockBody[FIELD_ID] = '11-22';
+
+      const result = numberHyphenSpacesOnlyValidation(mockBody, FIELD_ID, mockErrorMessagesObject, mockErrors, MINIMUM, MAXIMUM);
+
+      const expected = generateValidationErrors(FIELD_ID, mockErrorMessagesObject.BELOW_MINIMUM, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the provided value is above maximum length', () => {
+    it('should return validation errors', () => {
+      mockBody[FIELD_ID] = '11-22-33-44';
+
+      const result = numberHyphenSpacesOnlyValidation(mockBody, FIELD_ID, mockErrorMessagesObject, mockErrors, MINIMUM, MAXIMUM);
+
+      const expected = generateValidationErrors(FIELD_ID, mockErrorMessagesObject.ABOVE_MAXIMUM, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the provided value is at minimum/maximum', () => {
+    it('should return provided errors object', () => {
+      mockBody[FIELD_ID] = '11-22-33';
+
+      const result = numberHyphenSpacesOnlyValidation(mockBody, FIELD_ID, mockErrorMessagesObject, mockErrors, MINIMUM, MAXIMUM);
+
+      expect(result).toEqual(mockErrors);
+    });
+  });
+
+  describe('when the provided value does not have hyphens or spaces', () => {
+    it('should return provided errors object', () => {
+      mockBody[FIELD_ID] = '112233';
```

Comment:
Should this be valid?

Reply:
yes, it can either be a number, number with hyphen or number with spces

Reply:
It's returning validation errors if pure numbers?

Reply:
no, 12345, 12 34 5 and 1-2-3-4-5 are all valid

Reply:
After looking through the code, I understand now.

It wasn't clear to me that this test in particular was returning the provided errors object (which has no errors).

Can we update the describe blocks so that the valid and invalid scenarios are in a parent describe block?